Here's an example of failure output that I got from the current nixos-unstable version of awscli2, just for reference:

```
=================================== FAILURES ===================================
_______________ test_connect_to_selfsigned_with_tls_verify_false _______________

app = <SphinxTestApp buildername='linkcheck'>

    @pytest.mark.sphinx('linkcheck', testroot='linkcheck-localserver-https', freshenv=True)
    def test_connect_to_selfsigned_with_tls_verify_false(app):
        app.config.tls_verify = False
        with https_server(OKHandler):
            app.build()
    
        with open(app.outdir / 'output.json', encoding='utf-8') as fp:
            content = json.load(fp)
>       assert content == {
            "code": 0,
            "status": "working",
            "filename": "index.rst",
            "lineno": 1,
            "uri": "https://localhost:7777/",
            "info": "",
        }
E       assert {'code': 0, '...neno': 1, ...} == {'code': 0, '...neno': 1, ...}
E         Omitting 4 identical items, use -vv to show
E         Differing items:
E         {'status': 'broken'} != {'status': 'working'}
E         {'info': "HTTPSConnectionPool(host='localhost', port=7777): Read timed out. (read timeout=0.05)"} != {'info': ''}
E         Use -v to get more diff

tests/test_build_linkcheck.py:629: AssertionError
```

The "read timed out" makes a race condition particularly suspect in my opinion. Anyways, I can't do the official review/approval but this LGTM.

--- a/pkgs/tools/admin/awscli2/default.nix
+++ b/pkgs/tools/admin/awscli2/default.nix
@@ -13,6 +13,11 @@
 let
   py = python3 // {
     pkgs = python3.pkgs.overrideScope (final: prev: {
+      sphinx = prev.sphinx.overridePythonAttrs (prev: {
+        disabledTests = prev.disabledTests ++ [
+          "test_check_link_response_only" # fails on hydra https://hydra.nixos.org/build/242624087/nixlog/1
+        ];
+      });
       ruamel-yaml = prev.ruamel-yaml.overridePythonAttrs (prev: {
         src = prev.src.override {
           version = "0.17.21";

